ui-manchette-with-spacetimechart/src/stories/base-with-waypoint-menu.stories.tsx

```diff
@@ -69,6 +77,36 @@ const ManchetteWithSpaceTimeWrapper = ({
     selectedTrain
   );
 
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      // Avoid closing the menu when clicking on another waypoint
+      if (activeWaypointId && (event.target as HTMLElement).closest('.waypoint')) {
+        setIsClickOnWaypoint(true);
+      }
+      // Close the menu if the user clicks outside of it
+      if (!menuRef.current?.contains(event.target as Node)) {
+        setActiveWaypointId(undefined);
+      }
+    };
+
+    const toggleButton = document.querySelector('.toggle-mode div');
+    const zoomButtons = document.querySelectorAll('.zoom-buttons span');
```

Comment:
This is not safe to do in a `useEffect()`.

Reply:
Layout then ? 😖